backend/src/routes/notes.ts

```diff
@@ -34,15 +37,48 @@ interface NoteItem {
 ]
  */
 router.get("/notes/:id", async (req: Request, res: Response, next: NextFunction) => {
-  try {
-    console.log("Getting...");
+  try{
     const id = req.params.id;
     const note = await Note.findById(id);
     if (note == null) {
-      throw new Error();
+      return res.status(ServiceError.NOTE_WAS_NOT_FOUND.status)
+          .send(ServiceError.NOTE_WAS_NOT_FOUND.message);
     }
-    const notes: NoteItem[] = note.answers as NoteItem[];
-    notes.forEach((note_answer) => {
+    if (note.type==="post"){
+      const temp = await Session.findOne({postSession : id});
+      if(temp == null){
+        return res.status(ServiceError.SESSION_WAS_NOT_FOUND.status)
+            .send(ServiceError.SESSION_WAS_NOT_FOUND);
+      }
+      else {
```

Comment:
We don't need `else` here so can reduce one level of nesting

```
if(condition) {
    // throw error OR return
} else {
    // some
}

is equivalent to 

if (condition) {
    // throw error OR return
}
// some code
```

Similarly, 
```
try {
    if (condition) {
        // throw error OR return
    }
    // some code
} catch {
    // handle error
}
```

is same as 

```
try {
    if (condition) {
        // throw error OR return
    } else {
        // some code
    }
} catch {
    // handle error
}
```
Return / throw in if is key here because when if block is hit function terminates